kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/installation/kroxylicious/KroxyliciousApp.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.installation.kroxylicious;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.kroxylicious.systemtests.executor.Exec;
+import io.kroxylicious.systemtests.templates.kroxylicious.KroxyliciousConfigTemplates;
+import io.kroxylicious.systemtests.utils.TestUtils;
+
+import static org.awaitility.Awaitility.await;
+
+/**
+ * The type Kroxylicious app.
+ */
+public class KroxyliciousApp implements Runnable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KroxyliciousApp.class);
+    private final String clusterIp;
+    private final Thread thread;
+    private long pid;
+
+    /**
+     * Instantiates a new Kroxylicious app.
+     *
+     * @param clusterIp the cluster ip
+     */
+    public KroxyliciousApp(String clusterIp) {
+        this.clusterIp = clusterIp;
+        thread = new Thread(this, "kroxy");
+        thread.start();
+    }
+
+    public void run() {
+        LOGGER.info("Launching kroxylicious app");
+        Path parentPath = Path.of(System.getProperty("user.dir")).getParent();
+        final Path targetPath = parentPath.resolve("kroxylicious-app").resolve("target");
+
+        final Path startScript = resolveStartScript(targetPath);
+        final Path configFile = generateKroxyliciousConfiguration();
+        pid = Exec.execWithoutWait(startScript.toAbsolutePath().toString(), "-c", configFile.toAbsolutePath().toString());
+        while (!thread.isInterrupted()) {
```

Comment:
I'm not really understanding what benefit you are getting from spinning here?  Why not just let the run method return?